pkg/resourcetotelemetry/resource_to_telemetry.go

```diff
@@ -54,26 +73,57 @@ func WrapMetricsExporter(set Settings, exporter component.MetricsExporter) compo
 	if !set.Enabled {
 		return exporter
 	}
-	return &wrapperMetricsExporter{MetricsExporter: exporter}
+	return &wrapperMetricsExporter{MetricsExporter: exporter, MatchAttributes: set.Exclude}
 }
 
-func convertToMetricsAttributes(md pmetric.Metrics) pmetric.Metrics {
+func convertToMetricsAttributes(md pmetric.Metrics, filter MatchAttributes) (pmetric.Metrics, error) {
 	cloneMd := pmetric.NewMetrics()
 	md.CopyTo(cloneMd)
 	rms := cloneMd.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		resource := rms.At(i).Resource()
 
+		labelMap, err := extractLabelsFromResource(&resource, filter)
+		if err != nil {
+			return cloneMd, err
+		}
+
 		ilms := rms.At(i).ScopeMetrics()
 		for j := 0; j < ilms.Len(); j++ {
 			ilm := ilms.At(j)
 			metricSlice := ilm.Metrics()
 			for k := 0; k < metricSlice.Len(); k++ {
-				addAttributesToMetric(metricSlice.At(k), resource.Attributes())
+				addAttributesToMetric(metricSlice.At(k), labelMap)
 			}
 		}
 	}
-	return cloneMd
+	return cloneMd, nil
+}
+
+// extractAttributesFromResource extracts the attributes from a given resource and
+// returns them as a StringMap.
+func extractLabelsFromResource(resource *pcommon.Resource, filter MatchAttributes) (pcommon.Map, error) {
+	labelMap := pcommon.NewMap()
```

Comment:
I see you create a new set of resource attributes here and copy filtered ones there. So your solution is actually less performant than using the resource processor. You are doing 2 copy operations: on pmetric.Metrics and on resource attributes while resource processor makes only one copy of pmetric.Metrics

Reply:
It depends how many attributes is excluded, but in general you are right - there is a place to improvement. Anyway I was looking into the _resource processor_ and more advanced _attributes processor_ and transform processor I dont'n see any option to copy from resource attributes to metrics labels with potential exclusion. All options describes either modifications of resource attributes OR modifications of metrics labels themselves. Please correct me if I am wrong.

Reply:
Anyway refactored - only one copy operation is performed for attributes which not matching exclusion

Reply:
> I dont'n see any option to copy data from resource attributes to metrics labels with potential exclusion

I meant removing resource attributes with the resource processor and using the existing 
`pkg/resourcetotelemetry` functionality to move all of them into datapoint attributes. But I missed that it'll actually be two copy operations of the whole metrics in the result. 

I still don't think it's an ideal approach to rely on the soon-to-be-deprecated config interface in this package. It would be ideal if the transform processor can move resource attributes to the datapoint attributes. @TylerHelmuth do we have any plans to add such functionality in the transform processor?

Reply:
According to @TylerHelmuth you can use transform processor for this and use `pkg/resourcetotelemetry` at all.

You need to configure two following statements:
- delete_matching_keys(target, pattern)
- merge_maps(attributes, resource.attributes, "upsert")

Please try this out and let us know how it works.